### PR TITLE
Feature/renderer performance

### PR DESCRIPTION
**Pull request by ksyasuda**
## Summary

- **Subtitle Prefetching**: New `SubtitlePrefetchService` parses external subtitle files (SRT/VTT/ASS) and tokenizes upcoming lines in the background before they appear on screen, converting most cache misses into cache hits during normal playback
- **Batched Annotation Pass**: Collapsed 4 sequential `.map()` passes (known-word, frequency filtering, JLPT, N+1) into a single `.map()` + separate N+1 pass, eliminating 3 intermediate array allocations
- **DOM Template Pooling**: Replaced `createElement('span')` with `cloneNode(false)` from a pre-created template and `innerHTML = ''` with `replaceChildren()` to avoid HTML parser invocation

### Expected Impact

| Scenario | Before | After | Improvement |
|----------|--------|-------|-------------|
| Normal playback (prefetch-warmed) | ~200-320ms | ~30-50ms | ~80-85% |
| Cache hit (repeated subtitle) | ~72ms | ~55-65ms | ~10-20% |
| Cache miss (immediate seek) | ~200-320ms | ~150-260ms | ~20-25% |

## Test Plan

- [x] All 503 existing tests pass (1 pre-existing unrelated failure)
- [x] TypeScript type check passes with no errors
- [x] 18 new tests for subtitle cue parser (SRT, VTT, ASS formats, edge cases)
- [x] 9 new tests for prefetch service (priority window, cache-full stop, mid-flight stop, seek re-prioritization, pause/resume)
- [x] 3 new tests for SubtitleProcessingController (preCacheTokenization, isCacheFull)
- [ ] Manual testing: verify subtitle annotations appear faster during normal playback
- [ ] Manual testing: verify seek re-prioritizes prefetch window
- [ ] Manual testing: verify prefetch pauses during live subtitle processing

## Summary by CodeRabbit

* **New Features**
  * Background subtitle prefetching to tokenize and cache upcoming cues for smoother playback.
  * Unified parsing for SRT/VTT/ASS subtitle formats.

* **Performance**
  * Renderer uses template cloning and efficient DOM updates to cut rendering overhead.
  * Annotation switched to a batched single-pass flow, reducing allocations and passes.

* **Tests**
  * Extensive unit tests for parsing, prefetch logic, annotation, and renderer behaviors.

* **Documentation**
  * New architecture note describing renderer performance design and impact.

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -1071,6 +1079,81 @@ const subtitleProcessingControllerMainDeps = buildSubtitleProcessingControllerMa
...
+      },
+      isCacheFull: () => subtitleProcessingController.isCacheFull(),
+    });
+
+    subtitlePrefetchService.start(currentTimePos);
+    logger.info(`[subtitle-prefetch] started prefetching ${cues.length} cues from ${externalFilename}`);
+  } catch (error) {
+    logger.warn('[subtitle-prefetch] failed to initialize:', (error as Error).message);
+  }
+}
```
[the remark this answers is not in the document]
Comment (ksyasuda) [reply src/main.ts]: Fixed in d069df2. The async init path now goes through `src/main/runtime/subtitle-prefetch-init.ts` with a revision guard before attaching a service, and `src/main.ts` uses that controller for both refresh and cancel paths. Regression coverage: `src/main/runtime/subtitle-prefetch-init.test.ts`.
